keanu-project/src/main/java/io/improbable/keanu/util/Py4jByteArrayConverter.java

```diff
@@ -40,4 +41,53 @@
         byte[] byteArray = booleanBuffer.array();
         return byteArray;
     }
+
+    public static double[] toDoubleArray(byte[] byteArray, int itemSizeBytes) throws ArrayConversionException {
+        if (itemSizeBytes == 0) {
+            throw new ArrayConversionException("itemSizeBytes cannot be zero");
+        }
+
+        double[] doubles = new double[byteArray.length / itemSizeBytes];
+        for (int i = 0; i < doubles.length; i++) {
+            doubles[i] = ByteBuffer.wrap(byteArray, i*itemSizeBytes, itemSizeBytes).order(ByteOrder.LITTLE_ENDIAN).getDouble();
+        }
+        return doubles;
+    }
+
+    public static int[] toIntegerArray(byte[] byteArray, int itemSizeBytes) throws ArrayConversionException {
+        if (itemSizeBytes == 0) {
+            throw new ArrayConversionException("itemSizeBytes cannot be zero");
+        }
+
+        int[] ints = new int[byteArray.length / itemSizeBytes];
+        for(int i = 0; i < ints.length; i++) {
+            ints[i] = ByteBuffer.wrap(byteArray, i*itemSizeBytes, itemSizeBytes).order(ByteOrder.LITTLE_ENDIAN).getInt();
+        }
+        return ints;
+    }
+
+    public static boolean[] toBooleanArray(byte[] byteArray, int numberOfBooleansInArray) {
```

Comment:
why are the semantics different here (`numberOfBooleansInArray` instead of `itemSizeBytes`)? You're just passing in `byteArray.size * <size of a byte>`